### ISSUE TITLE
UI/Quest Translation in Korean 

### ISSUE THREAD

**Issue by merriweather04** [lokinmodar/Echoglossian#61]
While game dialog is able to render the Korean, UI/Journal are not able to display properly

**Comment by lokinmodar** [comment]
Hello! We will investigate this issue. It is probably related to the native game font and i have some ideas on how to solve this

**Comment by venafel** [comment]
Hi, i've got same issue with Polish langauge
![1](https://github.com/lokinmodar/Echoglossian/assets/169814534/17a340b8-d1fd-4f34-8471-cf9d7475b247)

**Comment by lokinmodar** [comment]
> Hi, i've got same issue with Polish langauge ![1](https://private-user-images.githubusercontent.com/169814534/330453825-17a340b8-d1fd-4f34-8471-cf9d7475b247.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2OTUxMjIsIm5iZiI6MTcxNTY5NDgyMiwicGF0aCI6Ii8xNjk4MTQ1MzQvMzMwNDUzODI1LTE3YTM0MGI4LWQxZmQtNGYzNC04NDcxLWNmOWQ3NDc1YjI0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNFQxMzUzNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01M2MyMWEyMWIwN2JlNTZiMzYyYWI5ZGM2OTljM2UzOWI4Y2Q0MjllNTNhYWU2ZGRhYWE3YjU4MGVjZTE2NzAwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.mOPaS-N7q2uU2UmWIUwsvoHXvM8c4fH0vAuvilSv5qg)

Yeah, i assume polish, czech, vietnamese and others will have this issue as well. The ingame font is missing the proper glyphs to render the special chars those languages use. I'm studying a way to help you but for now i recommend doing a Penumbra Font Mod. I don't have instructions for this but may be fairly easy to find in their discord...

**Comment by Sairef** [comment]
Hi! For the Ukrainian language, there is also an issue with fonts.
We have the mod for the UA symbols in chat (g4e_ukrchatsupport) and it does simply this:
![image](https://github.com/lokinmodar/Echoglossian/assets/23234835/27fb01fb-ef24-4747-87ee-63442fbe3953)
Maybe that could help :)

**Comment by lokinmodar** [comment]
> Hi! For the Ukrainian language, there is also an issue with fonts.
> 
> We have the mod for the UA symbols in chat (g4e_ukrchatsupport) and it does simply this:
> 
> ![image](https://github.com/lokinmodar/Echoglossian/assets/23234835/27fb01fb-ef24-4747-87ee-63442fbe3953)
> 
> Maybe that could help :)

Hello! Can you link me to where that code is? I want to take a deeper look into it. Thanks!

**Comment by Sairef** [comment]
> > Hi! For the Ukrainian language, there is also an issue with fonts.
> > We have the mod for the UA symbols in chat (g4e_ukrchatsupport) and it does simply this:
> > ![image](https://private-user-images.githubusercontent.com/23234835/331897893-27fb01fb-ef24-4747-87ee-63442fbe3953.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxNTAyNzQsIm5iZiI6MTcxNjE0OTk3NCwicGF0aCI6Ii8yMzIzNDgzNS8zMzE4OTc4OTMtMjdmYjAxZmItZWYyNC00NzQ3LTg3ZWUtNjM0NDJmYmUzOTUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE5VDIwMTkzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3MzY2YTRmZTQ0OTBlNDQzZmE2YjcyM2Q1ODkyNGEyMzAyNzM0ZTJlM2E3YTA3NmE4MWY5MjhmYWYzYzZiZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DJnAPL363Ag6ddfegaVpXUNtyqwYDA_7uFx2Q3KZf3k)
> > Maybe that could help :)
> 
> Hello! Can you link me to where that code is? I want to take a deeper look into it. Thanks!

https://github.com/justscribe/g4e_ukrchatsupport/blob/master/G4EUkrChatSupport/UkrChatSupport.cs#L267

**Comment by mderkowski82** [comment]
> Hi, i've got same issue with Polish langauge ![1](https://private-user-images.githubusercontent.com/169814534/330453825-17a340b8-d1fd-4f34-8471-cf9d7475b247.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4Mjk2OTcsIm5iZiI6MTcxNjgyOTM5NywicGF0aCI6Ii8xNjk4MTQ1MzQvMzMwNDUzODI1LTE3YTM0MGI4LWQxZmQtNGYzNC04NDcxLWNmOWQ3NDc1YjI0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyN1QxNzAzMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzAxYjFmMmIxOGYwZWQyOGQ3OWM3MTMwNjNjYzYyMGM5MDgwNGQzOGE5YjhiZGZhM2JiNGZjNjJjMzZmMWFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3AB0__ZnRXNFz0k9QT8RCRnqKewHJxQfpML2uE7NPXc)

use LC_ALL=C.UTF-8 in parameter

![image](https://github.com/lokinmodar/Echoglossian/assets/87711002/b0364e34-0a57-43ec-a336-fe3b5d486f14)

**Comment by AsymmetricParabola** [comment]
This is an issue with Thai, too. I'd be willing to contribute towards a bug bounty to get this resolved.

**Comment by lokinmodar** [comment]
I'm working on the new version. I intend to provide overlays for languages that the in-game font does not support

**Comment by lokinmodar** [comment]
Closing this as addressed in the current codebase.

The practical fix for unsupported native glyph languages was not to force more native text replacement, but to add quest/UI presentation paths that can use the plugin font stack instead of relying only on the game font. The newer overlay/hover-tooltip presentation paths now use the proper language font, which is how Korean and other unsupported-glyph languages are meant to be handled.

Native UI replacement is still constrained by the game font itself, but the plugin no longer depends on that path for quest-family readability.
